### PR TITLE
SONARIAC-1519 Add support for ConfigMap and Secret objects in cross-file analysis

### PR DESCRIPTION
**Pull request by rudy-regazzoni-sonarsource**

### CODE REVIEW

File: iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/checks/DuplicatedEnvironmentVariablesCheck.java [hunk 1]
```diff
@@ -23,6 +23,7 @@
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+
```
Comment (mstachniuk) [inline iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/checks/DuplicatedEnvironmentVariablesCheck.java:26]: There should be no such line, please check your IDE settings:
https://github.com/SonarSource/sonar-developer-toolset
```
Import Layout
* import all other imports
* <blank line>
* import static all other imports
```

File: iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/model/ConfigMap.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.kubernetes.model;
+
+import java.util.Map;
+
+import org.sonar.iac.common.yaml.tree.YamlTree;
+
+public record ConfigMap(
+  String path,
```
Comment (mstachniuk) [inline iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/model/ConfigMap.java:27]: The `filePath` will be a better name like in `ServiceAccount`.
I guess that the `name` property is missing here.
The same in `MapResource` and `Secret`.

File: iac-extensions/kubernetes/src/test/java/org/sonar/iac/kubernetes/visitors/ProjectResourceFactoryTest.java [hunk 3]
```diff
@@ -237,4 +244,108 @@ void shouldReturnNullForOtherKinds() {
...
+  @ParameterizedTest
+  @MethodSource("provideMapResource")
+  void shouldHandleKeyNotScalarInMapResource(String kindName, Class<? extends MapResource> clazz) {
+    // language=yaml
+    var code = """
+      apiVersion: v1
+      kind: %s
+      metadata:
+        namespace: my-namespace
+      data:
```
Comment (mstachniuk) [inline iac-extensions/kubernetes/src/test/java/org/sonar/iac/kubernetes/visitors/ProjectResourceFactoryTest.java:339]: Please add a test case where `data` is missing. The `data` is optional: https://kubernetes.io/docs/concepts/configuration/configmap/#configmap-object

**Review by mstachniuk — COMMENTED** [review]
Good progress, please don't forget about `name` property.

File: iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/checks/DuplicatedEnvironmentVariablesCheck.java (hunk 1 above)
Comment (rudy-regazzoni-sonarsource) [reply iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/checks/DuplicatedEnvironmentVariablesCheck.java:26]: Actually that's what I already have, not sure why it added a line here, I will check.
image

**Review by mstachniuk — APPROVED** [review]
LGTM

**Pull request merged**